docs: Microtica as a deployment option for Medusa Backend

naumovskamarija opened the pull request:
Contributing to the documentation with a new Medusa Backend Deployment Guide for Microtica - a cloud delivery platform that enables you to deploy infrastructure and applications to your AWS account.

shahednasser commented on docs/content/deployments/server/deploying-on-microtica.md:
```diff
@@ -0,0 +1,112 @@
+
+---
+
+## What will be provisioned on AWS
+Since Microtica deploys on your cloud account, here are the resources that the platform is going to provision in the environment.
+- VPC, subnets and networking
```
```suggestion
## What will be provisioned on AWS

Since Microtica deploys on your cloud account, here are the resources that the platform is going to provision in the environment.

- VPC, subnets and networking
```

shahednasser commented on docs/content/deployments/server/deploying-on-microtica.md:
```diff
@@ -0,0 +1,112 @@
+
+![Create Git Repository](https://global-uploads.webflow.com/62b04b3a715c202b5fa1408b/639727f93aa39723e1e5667f_RVoeExwARHAsVe3lq7oBZqYJ9nMP9x3EjX81evL3UgLZ_jXYeHk9mEBFJ760tQZvESp__mGaRKY6Nsd_NqELSzpNv3mWoqmfmJhosNlCLU94s3vhYKIhdo-T8sPvmtXZ2s6qPkkYdlHJ_2IWgDOvUn0uZzEnEl6beU65M5cmui2DZ8mkVi2wCquRBNIX.png)
+
+If you already have a Medusa server repo that you want to deploy then you need to perform a couple of changes in your source code. Follow [this tutorial to deploy and existing Medusa Server app](https://docs.microtica.com/medusa-server?utm_source=medusa&utm_medium=docs&utm_campaign=medusa#xUBRz).
```
```suggestion
If you already have a Medusa backend repository that you want to deploy then you need to perform a couple of changes in your source code. Follow [this tutorial to deploy and existing Medusa Backend app](https://docs.microtica.com/medusa-server?utm_source=medusa&utm_medium=docs&utm_campaign=medusa#xUBRz).
```

shahednasser commented on docs/content/deployments/server/deploying-on-microtica.md:
```diff
@@ -0,0 +1,112 @@
+
+## Deploy to Microtica
+
+### 1. Create/Import a Git repository
```
```suggestion
### Step 1: Create/Import a Git repository
```

shahednasser commented on docs/content/deployments/server/deploying-on-microtica.md:
```diff
@@ -0,0 +1,112 @@
+
+If you already have a Medusa server repo that you want to deploy then you need to perform a couple of changes in your source code. Follow [this tutorial to deploy and existing Medusa Server app](https://docs.microtica.com/medusa-server?utm_source=medusa&utm_medium=docs&utm_campaign=medusa#xUBRz).
+
+### 2. Configure the Template
```
```suggestion
### Step 2: Configure the Template
```

shahednasser commented on docs/content/deployments/server/deploying-on-microtica.md:
```diff
@@ -0,0 +1,112 @@
+
+### 2. Configure the Template
+
+The second step provides customization posibilities by configuring environment variables. Add an application name and the admin credentials **that will be used to create an initial admin user** with which you can later sign in to your Medusa Admin. 
```
```suggestion
The second step provides customization posibilities by configuring environment variables.

Add an application name and the admin credentials that will be used to create an initial admin user with which you can later sign in to your Medusa Admin. 
```

shahednasser commented on docs/content/deployments/server/deploying-on-microtica.md:
```diff
@@ -0,0 +1,112 @@
+
+The second step provides customization posibilities by configuring environment variables. Add an application name and the admin credentials **that will be used to create an initial admin user** with which you can later sign in to your Medusa Admin. 
+
+Choose whether you want a production Medusa Server environment or a development one. In production mode, this template will provision **managed RDS PostgreSQL and Redis instances**. In development mode, the template will use **local SQLite and a fake Redis instance**.
```
```suggestion
Then, choose whether you want a production Medusa Backend environment or a development one. The production template will provision your managed RDS PostgreSQL and Redis instances. On the other hand, the development template will use a local SQLite and a fake Redis instance.
```

shahednasser commented on docs/content/deployments/server/deploying-on-microtica.md:
```diff
@@ -0,0 +1,112 @@
+
+![Configure Template](https://global-uploads.webflow.com/62b04b3a715c202b5fa1408b/639727f9d62c7658915a3788_ERA-ETfPfTiCQ8wDPQZX6IxCZiDhX9GIj-jNTxIUfh6AKdC0BjdVKmuJbaaUlzQeITZWW00Ro1vFZj2R_arYBqD1AbWwlygPwzMOlwYGXby9ZtZ4pfI1A7jZ-oI4fmWHA3VmXtB1QR8NERVxEsT9R1zPQujSTIaWtQUs4gMAM6rVvA6iOfQZR7Qx1NcY.png)
+
+### 3. Connect an AWS account 
```
```suggestion
### Step 3: Connect an AWS account 
```

shahednasser commented on docs/content/deployments/server/deploying-on-microtica.md:
```diff
@@ -0,0 +1,112 @@
+
+### 3. Connect an AWS account 
+
+In the last step you select the environment in which you want to deploy the template. An existing default environment called `development` will be preselected here, or you can create a new environment.  
```
```suggestion
In the last step before deployment, select the environment in which you want to deploy the template. An existing default environment called `development` will be preselected here, or you can create a new environment.  
```

shahednasser commented on docs/content/deployments/server/deploying-on-microtica.md:
```diff
@@ -0,0 +1,112 @@
+
+![Connect AWS account](https://global-uploads.webflow.com/62b04b3a715c202b5fa1408b/639727f98144a7eb845dabee_GYZFu6wD5huAf6H5sCs4qpz09wQOaI2IMThsEd0CwR4PePc1ryF4mAlb4YkTZdIkX3Mnh_OzEr9bdm2r1_5GkYHHlFgHksP_hDj7DmlnitZmWaotj5th3R4wHK6gJzhIDmjfXeCk30YdK1kaeGH41JRCyteVH9y3YqzIyGz8xQlcDYKCoWVxDey47mPw.png)
+
+### 4. Deploy to AWS
```
```suggestion
### Step 4: Deploy to AWS
```

shahednasser commented on docs/content/deployments/server/deploying-on-microtica.md:
```diff
@@ -0,0 +1,112 @@
+
+![Building Resources](https://global-uploads.webflow.com/62b04b3a715c202b5fa1408b/639727f9a465c27cf7090e5f_gxwdXAAmUxHsSyJlgScHuu4m7GiFTNnwW-OZoAhDC8v-Ez3KmsK-OCKfF2Qe2JUtxMyih7jCA7n-hAZnx0fdFjv-TWyDuoAUvftSRE9hqmPB1gVhG6lH0IhXqbliou1VnAVDvlCtHobzlUrrDhnCmzob9PfdNGrzlhwsL60QsAXJm1gIPE27J6P_Y2Y3.png)
+
+Once the build process is complete, a new deployment with the infrastructure resources is triggered. You can follow the logs of the deployment process by clicking the **View deployment** button, and then selecting the deployment from the list. 
```
```suggestion
Once the build process is complete, a new deployment with the infrastructure resources is triggered. You can follow the logs of the deployment process by clicking the "View deployment" button, and then selecting the deployment from the list. 
```

shahednasser commented on docs/content/deployments/server/deploying-on-microtica.md:
```diff
@@ -0,0 +1,112 @@
+
+## Preview the environment
+
+After the deployment is finished, navigate to *Resources → [AppName] (Medusa in our case) → Overview*, and under *Resource Outputs* you should see the **AccessUrl**. This is the server url that you can use to access API endpoints. Try getting the list of products using the endpoint `/store/products`.
```
```suggestion
After the deployment is finished, navigate to Resources → [AppName] (Medusa in this example) → Overview. Then, under the Resource Outputs section you should see the AccessUrl. This is the backend's URL that you can use to access API endpoints. You can try getting the list of products using the endpoint `/store/products`.
```

shahednasser commented on docs/content/deployments/server/deploying-on-microtica.md:
```diff
@@ -0,0 +1,112 @@
+
+After the deployment is finished, navigate to *Resources → [AppName] (Medusa in our case) → Overview*, and under *Resource Outputs* you should see the **AccessUrl**. This is the server url that you can use to access API endpoints. Try getting the list of products using the endpoint `/store/products`.
+
+![Preview Medusa Server](https://global-uploads.webflow.com/62b04b3a715c202b5fa1408b/639727f9d423043b2f685a93_MsVHdJXNiV3oswGWvMAwZRG-ybWwm6QdwPynjQFNr6aFrKMCuG0oQZjxLkzct_dqkM6mLOdqi9SQJwiv_ZQJCPUNr15zTDBah3GFtT8r6EVlgAOS9ylJ0ObAVAfYMSuiXre25qz_4GywqeQT-8Ieh5CwYBTd_Y26jvZgUo75KbU8ftzFLjRoRxmU3GNu.png)
```
I think it's not necessary to include the image of the output. Maybe instead an image showcasing where to find the URL.

shahednasser commented on docs/content/deployments/server/deploying-on-microtica.md:
```diff
@@ -0,0 +1,112 @@
+
+## Add environment variables
+
+The environment variables can be updated/added/configured after deployment as well. You can access them with `process.env.<VARIABLE NAME>`. You can read more about the built-in environment variables, as well as how to specify custom environment variables in the Medusa Server runtime [in Microtica documentation](https://docs.microtica.com/medusa-server?utm_source=medusa&utm_medium=docs&utm_campaig ...
```
```suggestion
The environment variables can be updated, added, or configured after deployment as well. You can access them with `process.env.<VARIABLE NAME>`.

You can read more about the built-in environment variables, as well as how to specify custom environment variables in the Medusa backend runtime [in Microtica documentation](https://docs.microtica.com/medusa-server?utm_source=medusa&utm_medium=docs&utm_campaign=medusa#z8li6). 

```

shahednasser commented on www/docs/sidebars.js:
Seems like your setup changed the formatting of the file. Can you revert changes to only included the added sidebar item?

shahednasser reviewed (changes requested):
Thank you for your contribution @naumovskamarija ! Aside from the comments below there are additional points:

- Please rebase your PR to the `develop` branch. Yesterday we changed branches to make this the default one.
- Are there any troubleshooting guides that can be added here? If so, it would be great to add them in a new section at the end of the page.
- I'm assuming that any updates to the backend (for example, updating the Medusa core version) are deployed when changes are committed to the repository. Can you indicate how changes are deployed? Could be in a new section at the end of the documentation or added somewhere in the first step.

naumovskamarija commented:
Hi @shahednasser, I've added the suggested changes and updated the PR. All I have to do now is rebase to `develop`, but it appears that there are quite some changes between `develop` and `master`, specifically in the `www/docs/sidebars.js` file. So I'm wondering if it would be a better idea to start from scratch, fork the repo with `develop` as the default branch and then re-add the changes and open a new PR? Let me know how it's best to proceed.

shahednasser commented:
@naumovskamarija I'm fine with whatever is easier for you. If it's easier to start from scratch then you can do that 😇

naumovskamarija commented:
Okay then, I'll start over, and resubmit the PR again. Thanks for the support.
